src/main/java/graphql/schema/SchemaTransformer.java

```diff
@@ -100,23 +164,35 @@ public TraversalControl accept(TraverserContext<GraphQLSchemaElement> context, G
 
 
     /**
-     * Transforms a GrapQLSchema and returns a new GraphQLSchema object.
+     * Transforms a GraphQLSchema and returns a new GraphQLSchema object.
      *
      * @param schema
      * @param visitor
      *
      * @return a new GraphQLSchema instance.
      */
     public static GraphQLSchema transformSchema(GraphQLSchema schema, GraphQLTypeVisitor visitor) {
-        SchemaTransformer schemaTransformer = new SchemaTransformer();
-        return schemaTransformer.transform(schema, visitor);
+        return transformSchema(schema, TraversalType.ALL, visitor);
     }
 
+    public static GraphQLSchema transformSchema(GraphQLSchema schema, TraversalType traversalType, GraphQLTypeVisitor visitor) {
+        SchemaTransformer schemaTransformer = new SchemaTransformer();
+        return schemaTransformer.transform(schema, traversalType, visitor);
+    }
 
     public GraphQLSchema transform(final GraphQLSchema schema, GraphQLTypeVisitor visitor) {
+        return transform(schema, TraversalType.ALL, visitor);
+    }
+
+    public GraphQLSchema transform(final GraphQLSchema schema, TraversalType traversalType, GraphQLTypeVisitor visitor) {
 
+        DummyRootConfiguration rootConfiguration = new DummyRootConfiguration();
 
-        DummyRoot dummyRoot = new DummyRoot(schema);
+        if (traversalType == TraversalType.QUERY_ONLY) {
+            rootConfiguration.queryOnly();
```

Comment:
if we changed the name as suggested above then change it here

Reply:
Will do